src/backend/cuda/platform.cpp

```diff
@@ -670,6 +684,17 @@ DeviceManager::DeviceManager()
     AF_TRACE("Default device: {}", getActiveDeviceId());
 }
 
+DeviceManager::~DeviceManager() {
+    // Reset unique_ptrs for all cu[BLAS | Sparse | Solver]
+    // handles of all devices
+    for (int i = 0; i < nDevices; ++i) {
+        cublasManager(i).reset();
```

Comment:
I don't see you creating a custom destructor for the `unique_ptr`s. This will just calls free on the handles. Aren't you supposed to call the cu*(i.e. `cublasDestroy()`) functions to handle this? I would also imagine you need to call the `cudaSetDevice()` calls at each iteration.

Reply:
unique_ptrs manage objects of type https://github.com/arrayfire/arrayfire/blob/master/src/backend/cuda/cublas.hpp#L32
So, we don't need to call cu* calls directly.